src/main/java/dev/ebullient/convert/tools/pf2e/Json2QuteCreature.java

```diff
@@ -0,0 +1,145 @@
+    }
+
+    @Override
+    protected QuteCreature buildQuteResource() {
+        List<String> text = new ArrayList<>();
+        Tags tags = new Tags(sources);
+
+        appendToText(text, SourceField.entries.getFrom(rootNode), "##");
+
+        Collection<String> traits = collectTraitsFrom(rootNode, tags);
+        if (Pf2eCreature.alignment.existsIn(rootNode)) {
+            traits.addAll(toAlignments(rootNode, Pf2eCreature.alignment));
+        }
+        Optional<Integer> level = Pf2eCreature.level.getIntFrom(rootNode);
+
+        return new QuteCreature(sources, text, tags,
+                traits,
+                Field.alias.replaceTextFromList(rootNode, this),
+                Pf2eCreature.description.replaceTextFrom(rootNode, this),
+                level.orElse(null),
+                getPerception(),
+                buildDefenses(),
+                Pf2eCreatureLanguages.createCreatureLanguages(Pf2eCreature.languages.getFrom(rootNode), this));
+    }
+
+    /**
+     * <pre>
+     *     "perception": {
+     *         "std": 6
+     *     }
+     * </pre>
+     */
+    private Integer getPerception() {
+        JsonNode perceptionNode = Pf2eCreature.perception.getFrom(rootNode);
+        if (perceptionNode == null) {
+            return null;
+        }
+        return Pf2eCreature.std.getIntOrThrow(perceptionNode);
+    }
+
+    /**
+     * <pre>
+     *     "defenses": { ... }
+     * </pre>
+     */
+    private QuteDataDefenses buildDefenses() {
+        JsonNode defenseNode = Pf2eCreature.defenses.getFrom(rootNode);
+        if (defenseNode == null) {
+            return null;
+        }
+        return Pf2eDefenses.createInlineDefenses(defenseNode, this);
+    }
+
+    /**
+     * <pre>
+     *     "languages": {
+     *         "languages": ["Common", "Sylvan"],
+     *         "abilities": ["{&#64;ability telepathy} 100 feet"],
+     *         "notes": ["some other notes"],
+     *     }
+     * </pre>
+     */
+    enum Pf2eCreatureLanguages implements JsonNodeReader {
+        languages,
+        abilities,
+        notes;
+
+        static CreatureLanguages createCreatureLanguages(JsonNode node, Pf2eTypeReader convert) {
+            if (node == null) {
+                return null;
+            }
+            return new CreatureLanguages(
+                Pf2eCreatureLanguages.languages.getListOfStrings(node, convert.tui()),
+                Pf2eCreatureLanguages.abilities.getListOfStrings(node, convert.tui()).stream()
+                    .map(convert::replaceText).toList(),
+                Pf2eCreatureLanguages.notes.getListOfStrings(node, convert.tui()).stream()
+                    .map(convert::replaceText).toList());
+        }
+    }
+
+    /**
+     * @param languages The languages known (optional)
+     * @param notes     Language-related notes (optional)
+     * @param abilities Language-related abilities (optional)
+     */
+    public record CreatureLanguages(
```

Comment:
I try to keep objects/records that hold template data in the Qute package to keep all of the arrows pointing in one direction. My loose rule: Json* classes can reference Qute* classes, but Qute* classes can not reference Json* classes. 

So this should probably teleport to be an inner class of QuteCreature..

Reply:
You'll also need to verify that this class is present in a native build.